### PR TITLE
Dynamic export formats

### PR DESCRIPTION
**Pull request by PeterNerlich**
### Short description

Allow export to other formats supported by tablib / django-import-export
We were troubleshooting what looked like an encoding problem with feedback exported as CSV, and by the time we found that EXCEL was at fault and our CSVs were flawless, this was already half done. I figured I might as well clean it up and submit it as a low prio PR.

### Proposed changes

- Extend feedback export URL endpoint by `format` parameter
- Check if requested format is available through the dependency library
- Automatically determine mime type

### Side effects

- We just throw a ValueError if the format is not available. This is likely not the most elegant solution, but at the moment we hard-code which format can be requested through the UI anyway.

### Resolved issues

none

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

**Comment by MizukiTemma** [comment]
Is this to fix the problem reported [here](https://chat.tuerantuer.org/digitalfabrik/pl/ebbqa7bjo3fcpxcxoqzf4rmcxc)?

**Comment by PeterNerlich** [comment]
@MizukiTemma More or less. The fix is to trick EXCEL into actually reading it as UTF-8. But this was the related discussion that resulted in this PR, yes.

File: integreat_cms/cms/templates/feedback/region_feedback_list.html [hunk 1]
```diff
@@ -184,9 +184,12 @@ <h1 class="heading">
                                 {% translate "Delete" %}
                             </option>
                         {% endif %}
-                        <option data-bulk-action="{% url 'export_region_feedback' region_slug=request.region.slug %}">
+                        <option data-bulk-action="{% url 'export_region_feedback' region_slug=request.region.slug file_format='csv' %}">
                             {% translate "Export to CSV" %}
                         </option>
+                        <!--<option data-bulk-action="{% url 'export_region_feedback' region_slug=request.region.slug file_format='xlsx' %}">
```
Comment (JoeyStk) [inline integreat_cms/cms/templates/feedback/region_feedback_list.html:190]:
```suggestion

```
Since we agreed that xlsx shouldn't be supported, we can remove this and the comment below

File: integreat_cms/locale/de/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -5578,6 +5578,10 @@ msgstr "Noch kein Feedback vorhanden."
 msgid "Export to CSV"
 msgstr "Als CSV-Datei exportieren"
 
+#: cms/templates/feedback/region_feedback_list.html
+msgid "Export to XLSX"
```
Comment (JoeyStk) [inline integreat_cms/locale/de/LC_MESSAGES/django.po:5582]:
```suggestion
```
And we can also undo the translation for it

**Review by JoeyStk — COMMENTED** [review]
Thank you. I have two small comments before reviewing this properly :)

File: integreat_cms/core/management/commands/duplicate_feedback.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
```
Comment (JoeyStk) [inline integreat_cms/core/management/commands/duplicate_feedback.py:1]: As discussed this file should not be part of this PR :)

**Review by JoeyStk — APPROVED** [review]
Except from my comment below and resolving the failing tests, I don't have any further objections

**Comment by JoeyStk** [comment]
Ah, and one last information I forgot to mention. The length of your commit message is currently at 72 characters. I was wondering, if this might be to long. However the [Linux kernel documentation](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/process/submitting-patches.rst?id=bc7938deaca7f474918c41a0372a410049bd4e13#n664) says: 

> For these reasons, the ``summary`` must be no more than 70-75
characters, and it must describe both what the patch changes, as well
as why the patch might be necessary.  It is challenging to be both
succinct and descriptive, but that is what a well-written summary
should do.

So apparently this is not an issue. I just wanted to mention to you that it was something I thought about, and let you know what made me decided not to make it part of my review :)

File: integreat_cms/cms/views/feedback/region_feedback_actions.py [hunk 4]
```diff
@@ -204,8 +208,13 @@ def export_region_feedback(
         data = resource.export_resource(obj)
         dataset.append(data)
 
-    response = HttpResponse(dataset.csv, content_type="text/csv")
+    if file_format in map(lambda f: f.title, format_registry.formats()):
```
Comment (david-venhoff) [inline integreat_cms/cms/views/feedback/region_feedback_actions.py:211]:
```suggestion
    if file_format in (f.title for f in format_registry.formats()):
```
Since we recently had a pr removing all usages of map, I think we shouldn't use it anymore

**Review by david-venhoff — APPROVED** [review]
LGTM
